purchase_custom_search_product/views/purchase_view.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="purchase_order_form_inherited" model="ir.ui.view">
+            <field name="name">purchase.order.form.inherit</field>
+            <field name="model">purchase.order</field>
+            <field name="inherit_id" ref="purchase.purchase_order_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//notebook//page/field[@name='order_line']//field[@name='product_id']"
+                       position="attributes">
+                    <attribute name="domain">[('purchase_ok', '=', True)]</attribute>
```

Comment:
@remytms is there a way to add elements to a domain instead of overriding it?

Reply:
@robinkeunen perhaps this example for class : 
```xml
<attribute name="class" add="col-sm-5 o_grid" separator=" "/>
```
But for a domain it's not so easy. So I think there is no better way than replacing it. :/

Reply:
So, something to change ?

Reply:
No, it's ok. I checked `purchase.order.line` and there are no additional domain. In fact, it is already 
```
    product_id = fields.Many2one(... domain=[('purchase_ok', '=', True)] ... )
```
so I don't understand why you'd need to overload it.